webrecorder/browsertrix
# Create browser profile UI

(https://github.com/webrecorder/browsertrix-cloud/issues/199) Allows user to create browser profiles through the UI.

### Manual testing:
With a non-Chromium browser like Firefox:
1. Go to an archive -> "Browser Profiles"
2. Click "New Browser Profile". Verify that you see a warning message and cannot continue

With a Chromium browser like Chrome:
1. Go to an archive -> "Browser Profiles"
2. Click "New Browser Profile". Verify that dialog with starting URL form is shown. (fields under "more options") don't work as of this PR)
3. Fill in URL and click "Start Browser". Verify you're taken to the browser creation page and see a loading indicator
4. Click "Go Fullscreen". Verify fullscreen enters and exits as expected
5. Fill in name and description and click "Save Profile". Verify you're taken to the browser profiles list, and the new profile is visible

### Screenshots
**Button in list view:**
<img width="1041" alt="Screen Shot 2022-04-13 at 3 50 19 PM" src="https://user-images.githubusercontent.com/4672952/163284660-40a52d5e-d6b5-4ebd-a1bd-2f56b8a2f181.png">

**Start dialog:**
<img width="531" alt="Screen Shot 2022-04-13 at 3 50 27 PM" src="https://user-images.githubusercontent.com/4672952/163284662-cd007af6-51c2-4f43-891f-6d4eb2c11811.png">

**Start dialog in non-chromium browser:**
<img width="511" alt="Screen Shot 2022-04-13 at 4 43 12 PM" src="https://user-images.githubusercontent.com/4672952/163287272-4a7f30bd-8c12-470c-b0b8-a65e0a320cdf.png">


**Browser loading view:**
<img width="1473" alt="Screen Shot 2022-04-13 at 3 53 22 PM" src="https://user-images.githubusercontent.com/4672952/163284663-da9e6eca-7cbc-4765-b14c-95b4458498f2.png">

**Browser done loaded:**
<img width="1038" alt="Screen Shot 2022-04-13 at 3 53 42 PM" src="https://user-images.githubusercontent.com/4672952/163284664-78f61a49-4666-4459-8cce-892550a9ed3d.png">

**Fullscreen mode:**
![Screen Shot 2022-04-13 at 3 53 48 PM](https://user-images.githubusercontent.com/4672952/163284751-3fb39376-b1dd-4ebc-9bcb-9e66bfb39849.png)

**Error on invalid/timed out browser:**
<img width="1035" alt="Screen Shot 2022-04-13 at 3 11 11 PM" src="https://user-images.githubusercontent.com/4672952/163284655-5de90d15-28bf-4779-82de-8303f77dcd83.png">
